polyfill/lib/plaindatetime.mjs

```diff
@@ -35,22 +35,14 @@ export class PlainDateTime {
     isoYear = ES.ToIntegerWithTruncation(isoYear);
     isoMonth = ES.ToIntegerWithTruncation(isoMonth);
     isoDay = ES.ToIntegerWithTruncation(isoDay);
-    hour = ES.ToIntegerWithTruncation(hour);
-    minute = ES.ToIntegerWithTruncation(minute);
-    second = ES.ToIntegerWithTruncation(second);
-    millisecond = ES.ToIntegerWithTruncation(millisecond);
-    microsecond = ES.ToIntegerWithTruncation(microsecond);
-    nanosecond = ES.ToIntegerWithTruncation(nanosecond);
+    hour = hour === undefined ? 0 : ES.ToIntegerWithTruncation(hour);
+    minute = minute === undefined ? 0 : ES.ToIntegerWithTruncation(minute);
+    second = second === undefined ? 0 : ES.ToIntegerWithTruncation(second);
+    millisecond = millisecond === undefined ? 0 : ES.ToIntegerWithTruncation(millisecond);
+    microsecond = microsecond === undefined ? 0 : ES.ToIntegerWithTruncation(microsecond);
+    nanosecond = nanosecond === undefined ? 0 : ES.ToIntegerWithTruncation(nanosecond);
     calendar = ES.ToTemporalCalendar(calendar);
 
-    // Note: if the arguments are not passed,
-    //       ToIntegerWithTruncation(undefined) will have returned 0, which will
-    //       be rejected by RejectDateTime in CreateTemporalDateTimeSlots. This
-    //       check exists only to improve the error message.
-    if (arguments.length < 3) {
-      throw new RangeError('missing argument: isoYear, isoMonth and isoDay are required');
-    }
-
```

Comment:
You're right that this message is now dead code, and the comment would need to be changed, but maybe we could move it above the ToIntegerWithTruncation calls instead? (Let's do the same thing for the other similar messages in other types' constructors.) The RangeError would be thrown anyway by ToIntegerWithTruncation, so the difference is not observable, but this extra check improves the experience in the browser playground for a user who tries `new Temporal.PlainDateTime()` with no arguments.

Reply:
Um, If I move this check before the ToIntegerWithTruncation call, then we'll have to change the order-of-operations in missing-arguments tests.🤔

Reply:
Makes sense, I didn't think of that.

Reply:
We should probably remove the check from PlainDate, PlainYearMonth, and PlainMonthDay as well.